woocommerce/woocommerce-android
# Issue/2186 update product categories

Fixes #2186 by adding logic to display and update product categories. 

The changes in this PR are pulled from [this one](https://github.com/woocommerce/woocommerce-android/pull/2378) but made modifications to it since there has been a lot of changes from the time this PR was active.

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/83420974-057d0700-a445-11ea-9a42-15a5d06386de.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/83420970-03b34380-a445-11ea-987b-5c0b413f82cf.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/83421038-1af23100-a445-11ea-8806-5c64b65ffd0f.gif" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/83421044-1d548b00-a445-11ea-9c3c-0bb2f9e06b04.gif" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/83421101-2fcec480-a445-11ea-9909-1dc828ced3a4.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/83421110-30fff180-a445-11ea-8a4f-7c57dea70639.png" width="300"/>

#### To test
- Click on a product from the products tab and verify that the category of the product is displayed correctly. The categories will be displayed up to a max of 5 lines.
- Click on the categories section in the product and verify that the categories of the site are displayed correctly.
- Verify that pagination works correctly.
- Verify that sorting matches the UI in the web.
- Verify that you are able to select multiple categories from the list.
  - Notice that the `DONE` menu button is only visible when a new category is selected.
  - Click on the back menu button and notice that a discard dialog is displayed.
  - Click on the `Discard` button and notice that the changes you made are discarded in the product detail screen.
- Click on the categories section again.
  - Select a new category from the list.
  - Click on the `DONE` menu button and notice that the category you selected is displayed in the product detail screen.
  - Click on `UPDATE` menu button and verify that the category is successfully updated.
- Click on the categories section of a product.
  - De-select all the selected categories and click on the `DONE` menu button.
  - Click on the `UPDATE` menu button and verify that the category is successfully updated.


Update release notes:
